babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -788,15 +788,71 @@ SqlIndex SqlCreateTableIndex(Span s) :
 SqlNode SqlExecMacro() :
 {
     SqlIdentifier macro;
+    SqlNodeList paramNames = new SqlNodeList(getPos());
+    SqlNodeList paramValues = new SqlNodeList(getPos());
     Span s;
 }
 {
     macro = CompoundIdentifier() { s = span(); }
+    [
+        SqlExecMacroArgument(paramNames, paramValues)
+    ]
     {
-        return new SqlExecMacro(s.end(this), macro);
+        return new SqlExecMacro(s.end(this), macro, paramNames, paramValues);
     }
 }
 
+void SqlExecMacroArgument(SqlNodeList paramNames, SqlNodeList paramValues) :
```

Comment:
Does this function require that all params be of the same type (e.g. all with a named value or all without)? Is that a requirement according to the docs?

Reply:
On a related note, it would be good to add a test case where some of the params are named and others arent, for example: execute foo (1, secondparam = 'Hello')
Whether this should fail or pass depends on specs.

Reply:
This function does require all the params to be either all with a named value or all without. I would add more explanation in the design doc.